echoregions/lines/lines.py

```diff
@@ -98,48 +76,54 @@ def swap_val(val: Union[int, float]) -> Union[int, float]:
         if self.nan_depth_value is None:
             return
 
-        regions = self.data if inplace else self.data.copy()
+        regions = self._data if inplace else self._data.copy()
         regions.loc[:] = regions.apply(replace_depth, axis=1)
         return regions
 
-    def to_csv(self, save_path: str = None, **kwargs) -> None:
-        """Convert an EVL file to a CSV
+    def to_csv(self, save_path: bool = None) -> None:
+        """Save a Dataframe to a .csv file
 
         Parameters
         ----------
         save_path : str
-            Path to save csv file to
-        kwargs : keyword arguments
-            Additional arguments passed to `Lines.parse_file`
+            path to save the CSV file to
         """
-        if self.data is None:
-            self.parse_file(**kwargs)
-        self._parser.to_csv(self.data, save_path=save_path, **kwargs)
+        if not isinstance(self._data, DataFrame):
+            raise TypeError(
+                f"Invalid ds Type: {type(self._data)}. Must be of type DataFrame."
+            )
```

Comment:
since `._data` is a private attribute, it seems having the type checking here is redundant and somewhat strange, since the type is set by the programmer